src/vs/workbench/contrib/terminal/electron-browser/terminalInstance.ts

```diff
@@ -1036,15 +1062,19 @@ export class TerminalInstance implements ITerminalInstance {
 	}
 
 	private _attachPressAnyKeyToCloseListener() {
-		this._processManager!.addDisposable(dom.addDisposableListener(this._xterm.textarea, 'keypress', (event: KeyboardEvent) => {
+		const keyPressListener = dom.addDisposableListener(this._xterm.textarea, 'keypress', (event: KeyboardEvent) => {
+			keyPressListener.dispose();
 			this.dispose();
 			event.preventDefault();
-		}));
+		});
 	}
 
 	public reuseTerminal(shell: IShellLaunchConfig): void {
 		// Kill and clear up the process, making the process manager ready for a new process
-		this._processManager!.dispose();
+		if (this._processManager) {
+			this._processManager.dispose();
+			this._processManager = undefined;
```

Comment:
I don't think this is needed?

Reply:
@Tyriar - Yes. There is no process manager for a custom execution, there is no process to manage

Reply:
I meant setting it to undefined

Reply:
@Tyriar Is it possible that a terminal will be re-used from a "process" backed task to a "custom execution" (non process backed) task? In that case, I think we do want to set it to undefined.

Reply:
@GabeDeBacker I don't think that's possible